templates/CRM/Contribute/Form/Contribution/Main.extra.tpl

```diff
@@ -0,0 +1,46 @@
+{literal}
+  <script type="text/javascript">
+    CRM.$(function($) {
+      if (cj('#discountcode').length > 0) {
+        skipPaymentMethod();
+      }
+    });
+    function showHidePayment(flag) {
+      var payment_options = cj(".payment_options-group");
+      var payment_processor = cj("div.payment_processor-section");
+      var payment_information = cj("div#payment_information");
+      if (flag) {
+        payment_options.hide();
+        payment_processor.hide();
+        payment_information.hide();
+        // also unset selected payment methods
+        cj('input[name="payment_processor"]').removeProp('checked');
+      }
+      else {
+        payment_options.show();
+        payment_processor.show();
+        payment_information.show();
```

Comment:
these 3 variables are never acted on independently, and the if statement could be replaced by `.toggle` so all 14 lines of this could be reduced to:

    $('.payment_options-group, div.payment_processor-section, #payment_information).toggle(flag);